pkg/agent/proxy/proxier.go

```diff
@@ -662,17 +668,24 @@ func (p *proxier) installLoadBalancerService(svcInfoStr string,
 				IsExternal:         true,
 				IsNodePort:         false,
 				IsNested:           false, // Unsupported for LoadBalancerIP
-				IsDSR:              features.DefaultFeatureGate.Enabled(features.LoadBalancerModeDSR) && loadBalancerMode == agentconfig.LoadBalancerModeDSR,
+				IsDSR:              isDSR,
 			}); err != nil {
 				return fmt.Errorf("failed to install LoadBalancer load balancing flows: %w", err)
 			}
 			if p.proxyAll {
 				if err := p.addRouteForServiceIP(svcInfoStr, ip, p.routeClient.AddExternalIPRoute); err != nil {
 					return fmt.Errorf("failed to install LoadBalancer traffic redirecting routes: %w", err)
 				}
+				loadBalancerIPs = append(loadBalancerIPs, ip)
 			}
 		}
 	}
+	if p.proxyAll && len(loadBalancerIPs) != 0 {
+		if err := p.routeClient.AddLoadBalancerConf(loadBalancerIPs, svcPort, protocol, p.isIPv6); err != nil {
+			return err
+		}
+	}
```

Comment:
It seems it would call routeClient twice, one for installing route to IP, another for installing iptables but basically for the same purpose.

I feel it's duplicate and looks somewhat confusing. Maybe we should merge the two functions and move the route reference counter logic to routeClient.

Reply:
> It seems it would call routeClient twice, one for installing route to IP, another for installing iptables but basically for the same purpose.

You are right. I will try to merge such functions.

Reply:
Not finished the test code yet. Could you help review the code except for the test files in` pkg/agent/route`? @tnqn

Reply:
One more thing maybe we should reconsider, how about using the ipset that only stores Service IPs(LoadBalancerIPs or external IPs) instead of the tuples of `ServiceIP, protocol, servicePort` in the iptables rule to bypass kube-proxy? The current code is the later design. While I was coding it, I found that the former design can also work and the code may be clearer and readable. I would appreciate it if you could help estimate if there are any risks that we bypass Service IPs at the grain of IP.  @tnqn @antoninbas